packages/smooth_app/lib/pages/product_list_user_dialog_helper.dart

```diff
@@ -201,17 +175,62 @@ class ProductListUserDialogHelper {
     }
     return result;
   }
+
+  /// Shows all user lists with checkboxes, adds all [barcodes] to the selected lists.
+  /// Pre-checks all lists where [barcodes][0] is included if [barcodes.length] == 1
```

Comment:
Why only `barcodes[0]`?

Reply:
I honestly didn't want to introduce a second LoadingDialog for the potentially longer computation time of going through every product in every list and comparing them to the provided list.

Should I add this?

Reply:
It's not a matter of `LoadingDialog`: we're talking about almost instant local computing with the database, the performances are not an issue there (if needed we can improve the search in SQLite).
The _functional_ question is: why would that make sense for the user to see pre-checked lists if _only the first barcode_ is there? I would say "all barcodes" instead of just one. Or just would not care about that feature in a first approach: it's just a "nice to have".